src/highlighter.js

```diff
@@ -119,7 +136,23 @@ export default function (options) {
                     isWrappable(range.commonAncestorContainer) &&
                     !isWrappingNode(range.commonAncestorContainer.parentNode)
                 ) {
-                    range.surroundContents(getWrapper(currentGroupId));
+                    const wrapperNode = getWrapper(currentGroupId);
+                    addSplitData(wrapperNode, range.startOffset > 0, range.endOffset < range.commonAncestorContainer.length);
+                    const containerPreviousSibling = range.commonAncestorContainer.previousSibling;
+                    const containerNextSibling = range.commonAncestorContainer.nextSibling;
+                    range.surroundContents(wrapperNode);
+                    if (keepEmptyNodes) {
+                        //'surroundContents' can create empty text nodes, which will cause trouble in 'mergeAdjacentNodes' later
+                        if (wrapperNode.previousSibling && wrapperNode.previousSibling !== containerPreviousSibling
+                            && isText(wrapperNode.previousSibling) && wrapperNode.previousSibling.textContent.length === 0) {
+                            wrapperNode.previousSibling.remove();
+                        }
```

Comment:
Suggestion: A case for a small helper function? `removeEmptyTextNode(node, nodeToCompare)`